### DISCUSSION TITLE
Issue with gridify

### DISCUSSION THREAD

**Discussion by safee-group** [mtex-toolbox/mtex#1530] [Ask Anything]
Hello,

I am facing an issue with Gridify function. 

After using the Gridify data, plots become distorted. Attached two figures before and after using Gridify function (1_IQ - Before Gridify, 2_IQ - After Gridifying data). Also I would like to mention the plotted  EBSD scan data has two phases.

Any suggestions in this regard is really helpful.

Thank you in advance

Kind regards

![1_IQ](https://user-images.githubusercontent.com/91934764/185056820-9660d808-9ebf-421a-a6f0-e874eef44bc1.jpeg)
![2_IQ](https://user-images.githubusercontent.com/91934764/185056844-bf9d516b-675c-4f7b-a104-74e6e9b9f52b.jpeg)

**Comment by nyyssont** [comment]
It looks like you are gridifying hexagonal data. When converting to a square grid, the closest corresponding point in the hex grid will be selected. This might lead to loss of data or small distortions.

Read more here:
https://mtex-toolbox.github.io/EBSDInter.html

**Comment by safee-group** [comment]
Hello 
Thank you for the reply.

I would like to mention to that even in case of Fig 2 in above post(2_IQ - After Gridifying data) the pixels are of rectangular rather than square. Can you suggest something to fix this.

Also, Could you please suggest me a method where I can retain the hexagonal grid even after the gridifying if possible.

Thank you in advance

**Comment by Tijmenvermeij** [comment]
Shouldn't gridify automatically detect that the data is hexagonal, and thus create an "EBSDhex" variable instead of an "EBSDsquare" variable? 
See here: https://mtex-toolbox.github.io/EBSD.gridify.html

Perhaps the calculated "unitCell" does not exactly correspond to hexagonal?

**Comment by safee-group** [comment]
Hello,

Yes gridify detects the original grid as hexagonal and is creating ESBD hex variable too. 

If I does gridify it on a small area (please see figure 1) it is retaining the hexagonal grid. Whereas, for large areas (figure 2) the pixel become a rectangular rather than hexagonal. I guess that it is issues with uneven spacing in x and y probably. But, I am not able figure out the solution for this. Any suggestion would be really appreciated. 


Thank you in advance

![Figure 1](https://user-images.githubusercontent.com/91934764/185894320-5d6c328f-0b44-4f64-92af-d10ffe31f726.jpeg)
![Figure 2](https://user-images.githubusercontent.com/91934764/185894327-53fac697-5a98-4856-af77-c83e57c3d6db.jpeg)

**Comment by kilir** [comment]
Hi,
what you are observing is due to mtex simplifying the plotting routine for large maps. If you like to plot the unitcells, add the option `'exact'` to plot(). Also see help `'EBSD/plot'`.
Hope this helps.
Cheers,
Rüdiger

**Comment by safee-group** [comment]
Hello 

Thank you for answer. It worked. 

Many thanks to all of you for the support. 

Kind regards
